[Improvement] In MetricsSystem.java use LinkedList

### What would you like to be improved?

MetricsSystem.java:38: warning: [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
  private List<Reporter> metricsReporters = new LinkedList<>();

See https://errorprone.info/bugpattern/JdkObsolete)

### How should we improve?

_No response_

--- a/core/src/main/java/com/datastrato/gravitino/metrics/MetricsSystem.java
+++ b/core/src/main/java/com/datastrato/gravitino/metrics/MetricsSystem.java
@@ -19,9 +19,9 @@ import io.prometheus.client.dropwizard.samplebuilder.MapperConfig;
 import io.prometheus.client.exporter.MetricsServlet;
 import java.io.Closeable;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.HashMap;
-import java.util.LinkedList;
 import java.util.List;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -35,7 +35,7 @@ public class MetricsSystem implements Closeable {
   private final String name;
   private final MetricRegistry metricRegistry;
   private HashMap<String, MetricsSource> metricSources = new HashMap<>();
-  private List<Reporter> metricsReporters = new LinkedList<>();
+  private List<Reporter> metricsReporters = new ArrayList<>();
   private CollectorRegistry prometheusRegistry;
 
   public MetricsSystem() {

